tests/shell/main.rs

```diff
@@ -27,6 +27,26 @@ fn loads_env() {
     );
 }
 
+#[test]
+fn path_with_spaces() {
+    let exemplar = "/opt/hello world/bin/:/mnt/c/Program Files (x86)/QuickTime/QTSystem:";
+    let new_path = std::env::var("PATH")
+        .map(|p| format!("{}:{}", p, exemplar))
+        .unwrap_or_else(|_| String::from(exemplar));
+    std::env::set_var("PATH", new_path);
```

Comment:
If I'm understanding it correctly, then what's important in this test is that the environment is set for the command (on line 39), right?

If so, my suggestion would be to remove the side-effect of changing this processes' environment and set the environment of the sub-process launched on line 39 instead, see https://doc.rust-lang.org/std/process/struct.Command.html#method.env.

I think it is then not necessary to manually query the current `"PATH"` and prepend it to the example. But I'm not 100% sure this works.

Just to be sure: can you confirm that this test fails without the changes to `envrc.bash`?

Reply:
You're correct, the PATH var is the important bit of the test.

I'll take a stab at modifying the environment of just the process here. I think I'll still need to append my exemplars to the current path so that `echo` works.

Reply:
Works without appending to the existing PATH 🙏 

Undo-ing my changes to `envrc.bash` breaks the test (and also `trivial_v1` and `trivial_v2` since they reference my current `PATH` which includes spaces)

Reply:
Which, now that I'm looking at it again, makes me realize that the test really belongs in `envrc.rs`, I'll rewrite it there.